### ISSUE TITLE
isMethod isn't working

### ISSUE THREAD

**Issue by lukaspj** [GarageGames/Torque3D#1106]
```C
$obj = new SimObject(){};
$obj.isMethod("getName"); // Returns true.
isMethod($obj, "getName"); // Returns false.
```
[this](https://github.com/GarageGames/Torque3D/blob/development/Engine/source/console/consoleFunctions.cpp#L2230-L2243) is the method in question.

```c++
   Namespace* ns = Namespace::find( StringTable->insert( nameSpace ) );
   Namespace::Entry* nse = ns->lookup( StringTable->insert( method ) );
   if( !nse )
      return false;

   return true;
```
It seems that 
```c++
Namespace* ns = Namespace::find( StringTable->insert( nameSpace ) );
```
Returns the namespace for the instance and not the "SimObject" namespace, the instance doesn't have any methods defines (I guess there's only variables in that namespace?), should it return the SimObject namespace instead?
Edit: plottwist:
```C
$obj = new SimObject(ASD){};
$obj.isMethod("getName"); // Returns true.
isMethod($obj, "getName"); // Returns false.
isMethod(ASD, "getName"); // Returns true.
```
So the id-namespace is the thing that goes wrong, seems like the named-namespace and the unnamed-namespace don't map to the same thing.

**Comment by lukaspj** [comment]
If you create a named object, there will be 2 namespaces:

* One for the ID
* One for the name

The named-namespace is correct, with the parent set and everything, the ID-namespace only has the name set to the id and points to the next namespace. No information about the parent or anything.

**Comment by jamesu** [comment]
"isMethod" the **function** is only supposed to check if a method exists within a **namespace**. It's not supposed to inspect an object.

"isMethod" the **object method** checks the **objects namespace**.

Based on the intended behaviour of both methods, there is nothing wrong with your code. It is working fine.

The only problem I see with the behaviour of the existing function is it will create a namespace if one doesn't exist. This doesn't seem like a good idea as you could potentially create a lot of invalid blank namespaces this way which could impact on performance.

**Comment by lukaspj** [comment]
@jamesu yeah just came to the same conclusion, and yes it's odd it creates unnecessary namespaces. Side-effects are bad .

Anyways, I'll close this as the issue wasn't an actual issue, rather a result of ambiguous function naming..

**Comment by eightyeight** [comment]
Retargeted this. Side-effects bad; let's see if we can fix it. We also may have a documentation issue that needs investigating.
